### PR TITLE
`Alert` and `Toast` Overview Update

### PR DESCRIPTION
**Pull request by andgen404**
### :pushpin: Summary
`Alert` and `Toast` - Changed the overview section for both components to have a bit more clarity on when to use. 

### 📸 Screen Shots 
[Current](https://helios.hashicorp.design/components/alert) documentation for `Alert` and `Toast`

Screenshot 2023-10-19 at 11 27 40 AM
<div> </div>
Screenshot 2023-10-19 at 11 28 07 AM
<div> </div>
Screenshot 2023-10-19 at 11 28 38 AM
<div> </div>
Screenshot 2023-10-19 at 11 27 52 AM

Preview 👉  👉  👉  👉  [Updated](https://hds-website-git-ag-alert-toast-documentation-update-hashicorp.vercel.app/components/alert) documentation for `Alert` and `Toast`

### :link: External links

Jira ticket: [HDS-2441](https://hashicorp.atlassian.net/browse/HDS-2441)

***

### 👀 Component checklist

- [ ] Percy was checked for any visual regression
- [ ] A11y tests have been run locally (`yarn test:a11y --filter="COMPONENT-NAME"`)
- [ ] If documenting a new component, an acceptance test that includes the `a11yAudit` has been added
- [ ] A changelog entry was added via [Changesets](https://github.com/changesets/changesets) if needed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

[HDS-2441]: https://hashicorp.atlassian.net/browse/HDS-2441?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: website/docs/components/alert/partials/guidelines/overview.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-Use alerts to display brief but important system-generated messages. They attract the user’s attention without interrupting their task. For messages that are the result of a user’s actions, see [Toast](/components/toast).
+Alerts display brief but important system-generated messages and capture the user's attention without interrupting their workflow. They excel in conveying crucial information or errors, such as validation errors that require user attention or action. Unlike [Toast](/components/toast), tailored for feedback on user actions, Alerts serve as the go-to component for more critical notifications.
```
Comment (heatherlarsen) [inline website/docs/components/alert/partials/guidelines/overview.md:1]:
```suggestion
Alerts display brief but important messages that capture the user's attention but don't interrupt their workflow. 
```

Per our discussion this morning, this content is better suited for the "Critical alerts" section between the heading and the first "do" block.

> Use critical alerts to convey crucial information or errors, such as validation errors requiring user action.

File: website/docs/components/toast/partials/guidelines/overview.md [hunk 2]
```diff
@@ -1 +1 @@
-A Toast is an element intended for messages resulting from a user’s actions. It typically disappears after a set interval or due to a user dismissing it. For system-generated messages, see the [Alert](/components/alert) component.
\ No newline at end of file
+Toasts deliver messages resulting from a user's actions. They typically disappear after a set interval or upon user dismissal, offering a non-intrusive way to provide feedback. While [Alerts](/components/alert) handle system-generated messages or critical information, Toasts suit less urgent feedback, such as indicating a user-initiated process's success or ongoing status.
```
Comment (heatherlarsen) [inline website/docs/components/toast/partials/guidelines/overview.md:1]: Providing feedback based on our earlier conversation. I removed the second half of this section to:

1. reduce confusion around terms such as "system-generated"
2. remove duplicative information (info is also presented in the when to use/when not to use section)

```suggestion
Toasts deliver messages to communicate an ongoing process or as a result of a user's actions. They typically disappear after a set interval or upon user dismissal, offering a non-intrusive way to provide feedback. 
```

**Review by heatherlarsen — CHANGES_REQUESTED** [review]
Great first try at updating the docs! I'm not sure they get to the root of the issue outlined in the Jira ticket, though, so I left a few comments to address that further. 

A few other ideas we discussed earlier for the `Alert` page include:

- move the "Usage" section above the "Types of alerts" section
- update the sentence "To communicate feedback on a user’s action..." to "To communicate successful feedback on a user’s action..."

File: website/docs/components/alert/partials/guidelines/overview.md [hunk 3]
```diff
@@ -1,4 +1,4 @@
-Use alerts to display brief but important system-generated messages. They attract the user’s attention without interrupting their task. For messages that are the result of a user’s actions, see [Toast](/components/toast).
+Alerts display brief but important messages that capture the user's attention but don't interrupt their workflow.
```
Comment (andgen404) [inline website/docs/components/alert/partials/guidelines/overview.md:1]:
```suggestion
Alerts display brief but important messages that capture the user's attention but doesn't interrupt their workflow. 
```

File: website/docs/components/alert/partials/guidelines/guidelines.md [hunk 4]
```diff
@@ -10,6 +10,22 @@
...
+
+Use page-level alerts on the top of the page, between the navigation and the breadcrumb to inform users of an event that may require the user’s attention or action.
+
+### Inline
+
+Use inline alerts when the message is contextual or specific to the section it’s placed in.
+
+### Compact
+
+Use compact alerts when a less prominent alert is needed.
```
[the remark this answers is not in the document]
Comment (heatherlarsen) [reply website/docs/components/alert/partials/guidelines/guidelines.md:27]: @MelSumner This is current content on the website, and changing it is out of the scope of the ticket. I asked Andrew to re-organize some of the content (moving the "Usage" section to the top), which is in scope.

File: website/docs/components/alert/partials/guidelines/guidelines.md [hunk 5]
```diff
@@ -6,10 +6,26 @@
 
 ### When not to use
 
-- To communicate feedback on a user’s action, consider [Toast](/components/toast).
+- For subtle feedback on a user’s action, consider [Toast](/components/toast).
```
Comment (andgen404) [inline website/docs/components/alert/partials/guidelines/guidelines.md:9]:
```suggestion
- To communicate successful feedback on a user’s action, consider [Toast](/components/toast).
```

File: website/docs/components/toast/partials/guidelines/overview.md [hunk 6]
```diff
@@ -1 +1 @@
-A Toast is an element intended for messages resulting from a user’s actions. It typically disappears after a set interval or due to a user dismissing it. For system-generated messages, see the [Alert](/components/alert) component.
\ No newline at end of file
+Toasts deliver messages to communicate an ongoing process or as a result of a user's actions. They typically disappear after a set interval or upon user dismissal, offering a non-intrusive way to provide feedback.
```
Comment (jorytindall) [inline website/docs/components/toast/partials/guidelines/overview.md:1]:
```suggestion
Toasts deliver messages to communicate an ongoing process or result of a user action. They typically disappear after a set interval or upon user dismissal, offering a non-intrusive way to provide feedback. 
```

File: website/docs/components/toast/partials/guidelines/overview.md [hunk 7]
```diff
@@ -1 +1 @@
-A Toast is an element intended for messages resulting from a user’s actions. It typically disappears after a set interval or due to a user dismissing it. For system-generated messages, see the [Alert](/components/alert) component.
\ No newline at end of file
+Toasts deliver messages to communicate an ongoing process or result of a user action. They typically disappear after a set interval or upon user dismissal, offering a non-intrusive way to provide feedback.
```
Comment (heatherlarsen) [inline website/docs/components/toast/partials/guidelines/overview.md:1]: Slight change to communicate that toasts may not always be the right component to communicate a result of a user action. wdyt?

```suggestion
Toasts deliver messages to communicate an ongoing process or, sometimes, a result of a user action. They typically disappear after a set interval or upon user dismissal, offering a non-intrusive way to provide feedback. 
```

**Review by heatherlarsen — APPROVED** [review]
Thanks for making those adjustments!

**Review by alex-ju — APPROVED** [review]

**Pull request merged**